runtime/README.md

```diff
@@ -106,56 +108,60 @@ The contexts contain the internal structures necessary to resume operations to s
 
 These contexts are intended to be opaque to the user, and SHALL be encrypted and authenticated if they contain sensitive internal data.
 
-### Data Storage
-
-The CM system has its own dedicated storage in DCCM.
-
-This storage is protected with authenticated encryption using internal keys that are randomly generated at power on and stored in the key vault.
-
-This storage is exposed to users through opaque 256-bit handles, called cryptographic mailbox IDs (CMIDs). These are mapped internally into DCCM storage blocks.
-
-Keys can be imported using `CM_IMPORT` and deleted via `CM_DELETE`.
-
-The entire contents can be cleared with `CM_CLEAR`.
+### Keys
 
-The status of the data storage can be queried with `CM_STATUS`.
+Cryptographic Mailbox Key (CMKs) are used to store keys. Certain commands generate and return a new CMK. Most commands that use CMKs will also return a new CMK, as it is necessary to track CMKs so that they are not used beyond any relevant limits for their key type.
 
-### Storage Design
+They are returned from commands that generate keys and must be passed back to Caliptra to be used. These keys are encrypted and opaque to the mailbox caller.
 
-Data is stored in 16-byte blocks.
+Internally, the unecrypted CMKs have the following structure:
 
-Each block's status is tracked in a bitfield, 1 bit per block, with a value of 1 indicating that block is in use.
+| **Name**      | **Bits** | **Description**                             |
+| ------------- | -------- | ------------------------------------------- |
+| version       | 32       | CMK version. Currently always 1.            |
+| key usage     | 32       | represents which kind of key this is        |
+| id            | 32       | ID number                                   |
+| usage counter | 64       | how many times this key has been used       |
+| length        | 32       | how many bits of key material are used      |
+| padding       | 64       | reserved for now                            |
+| key material  | 512      | bits used for the key material              |
 
-For example, if we have 32 KB of storage for CM, this requires 256 bytes of additional storage.
+The encrypted CMKs have the structure:
 
-### CMIDs
+| **Name**      | **Bits** | **Description**                |
+| ------------- | -------- | ------------------------------ |
+| padding       | 32       | reserved                       |
```

Comment:
I'm anticipating a case where we would want a CMK with different durability properties, like one which survives reset and could be stored in flash by the SoC manager. A field here that lets Caliptra distinguish which key to use when decrypting the CMK would be useful. Perhaps `domain`, and a 128-bit `domain_metadata` field, which holds domain-specific data.

The use-case for the latter field would be if we model an encrypted OCP L.O.C.K. PMEK as a CMK, where `domain` would indicate that the CMK is encrypted using a durable key derived from the storage root key and a provisioned access key, and `domain_metadata` is a truncated hash of the access key.

We could of course just come up with a separate data structure for that use-case. Up to you.

Reply:
I like it. Let me see if I can fit that in without increasing the size.